snappy/app/netfilter.go

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snappy
+
+import (
+	"log"
+	"net"
+	"net/http"
+)
+
+// NetFilter manages an IP-based filter to limit access to Snapweb
+type NetFilter struct {
+	AllowedNetworks []*net.IPNet
```

Comment:
I am not sure you need to export this field, ... if you do then the cache stuff might break since one can mutate AllowedNetworks manually and the cache will not be invalidated,